tests/Unit/Domain/CoordinateMaps/TimeDependent/Test_Shape.cpp

```diff
@@ -0,0 +1,462 @@
+
+  // this part essentially duplicates the code from the map
+  const auto spatial_part = transition_func(centered_coords);
+  const auto spatial_gradient = transition_func.gradient(centered_coords);
+  const auto spatial_part_over_radius =
+      transition_func.map_over_radius(centered_coords);
+  tnsr::Ij<DataVector, 3, Frame::NoFrame> result(num_points);
+  for (size_t i = 0; i < 3; ++i) {
+    for (size_t j = 0; j < 3; ++j) {
+      result.get(i, j) =
+          -gsl::at(centered_coords, i) *
+          (angular_part * gsl::at(spatial_gradient, j) +
+           gsl::at(cartesian_gradient, j) * spatial_part_over_radius);
+    }
+    result.get(i, i) += 1. - angular_part * spatial_part;
+  }
+  return result;
+}
+
+template <typename TransitionFunction>
+void test_analytical_jacobian(const TransitionFunction& transition_func,
+                              size_t l_max, size_t m_max, size_t num_points,
+                              gsl::not_null<std::mt19937*> generator) {
+  std::uniform_real_distribution dist{-10., 10.};
+  const auto center =
+      make_with_random_values<std::array<double, 3>>(generator, dist, 3);
+  auto target_data = make_with_random_values<std::array<DataVector, 3>>(
+      generator, dist, num_points);
+
+  // the analytical solution divides by sin(theta), so it might get into
+  // trouble at the poles, although it appeared fine without this safety
+  // measure in >10^8 tries. Here, the y-coordinate is increased if the point
+  // has theta close to zero.
+  const auto centered_coords = target_data - center;
+  const DataVector sin_thetas = sin(
+      atan2(hypot(centered_coords[0], centered_coords[1]), centered_coords[2]));
+  for (size_t i = 0; i < sin_thetas.size(); ++i) {
+    if (abs(gsl::at(sin_thetas, i)) < 1e-4) {
+      target_data[1][i] += 1.;
+    }
+  }
+
+  const auto random_coefs = generate_random_coefs(l_max, m_max, generator);
+
+  FunctionsOfTimeMap functions_of_time{};
+  double time{};
+  auto map = CoordinateMaps::TimeDependent::Shape{};
+  const auto spherepack_coefs =
+      convert_coefs_to_spherepack(random_coefs, l_max, m_max);
+  generate_random_map_time_and_f_of_time<TransitionFunction>(
+      make_not_null(&map), make_not_null(&time),
+      make_not_null(&functions_of_time), l_max, m_max, center, transition_func,
+      spherepack_coefs, true, generator);
+  const auto mapped_jacobian =
+      map.jacobian(target_data, time, functions_of_time);
+  const auto analytical_jacobian = calculate_analytical_jacobian(
+      target_data, center, l_max, m_max, random_coefs, transition_func);
+  CHECK_ITERABLE_APPROX(mapped_jacobian, analytical_jacobian);
+}
+
+}  // namespace
+
+SPECTRE_TEST_CASE("Unit.Domain.CoordinateMaps.TimeDependent.Shape",
+                  "[Domain][Unit]") {
+  domain::CoordinateMaps::register_derived_with_charm();
+
+  MAKE_GENERATOR(generator);
+  {
+    const CoordinateMaps::SphereTransition sphere_transition{1e-7, 100.};
```

Comment:
why is r_min so small and r_max so big?

Reply:
it is in theory possible that one of the random points used to test the shape map is outside of the domain of the transition function (inside the spherical "excised" shell). Making the inner radius so small reduces this probability to essentially zero for our purposes. I guess I should technically randomly pick the points only inside the transition function domain but this makes the generating the points more complicated.